Added bankers algorithm in java

shraiyya opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Pull request corresponding to issue number #1806 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Please add the DWOC tag.

shraiyya committed:
Added bankers algorithm in java

shraiyya committed:
Update README.md

shraiyya committed:
Updated Readme

HarshCasper commented on Java/Operating System/banker.java:
```diff
@@ -0,0 +1,338 @@
+import java.util.Scanner;
```
Change the Directory name from `Operating System` to `Operating-System`

HarshCasper commented on Java/Operating System/banker.java:
```diff
@@ -0,0 +1,338 @@
+import java.util.Scanner;
+
+class algorithm //class for calculation and display
```
Comments don't have to be very layman. We all understand what work does a class do.

HarshCasper commented on Java/Operating System/banker.java:
```diff
@@ -0,0 +1,338 @@
+ 
+void display(int r,int p,int[]avail,int max[][],int need[][],int allocation[][],int res[])
+{
+	System.out.println("THE MAXIMUM RESOURCES MATRIX IS AS FOLLOWS :");
```
Follow a consistent indentation

HarshCasper commented on Java/Operating System/banker.java:
```diff
@@ -0,0 +1,338 @@
+    System.out.println(" ");
+    System.out.println(" ");
+    
+for (int i = 0 ; i < p ; i++) {
```
Inconsistent indentation!

HarshCasper commented on Java/Operating System/banker.java:
```diff
@@ -0,0 +1,338 @@
+    	
+        for (int j = 0 ; j < r ; j++) {
+  
+            need[i][j] = max[i][j] - allocation[i][j]; // Need of instance = maxm instance -allocated instance 
```
Use comments above the line of code, not beside them

shraiyya committed:
Update and rename Java/Operating System/banker.java to Java/Operating-System/banker.java

shraiyya committed:
Changed the alignment of comments

shraiyya commented:
I am done with the changes. Please review it again.

shraiyya committed:
Merge pull request #1 from TesseractCoding/master

Updating My fork

shraiyya committed:
Create Huffman.java

shraiyya committed:
Update Huffman.java

shraiyya committed:
Update README.md

atarax665 reviewed (changes requested):
Remove the Huffman encoding program, you may have a separate PR for this.

shraiyya committed:
Delete Huffman.java

shraiyya committed:
Update README.md

shraiyya committed:
Merge pull request #3 from TesseractCoding/master

Updation of my fork

iamrajiv commented on Java/README.md:
```diff
@@ -179,6 +179,9 @@ _add list here_
 _add list here_
 
+## Operating Systems
+-  [Banker's Algorithm](Operating%20System/banker.java)
```
@shraiyya This link is not working. Update this to  `-  [Banker's Algorithm](Operating-System/banker.java)`.

iamrajiv commented on Java/Operating-System/banker.java:
```diff
@@ -0,0 +1,311 @@
+import java.util.Scanner;
```
@shraiyya Add description about Banker's algorithm.

iamrajiv commented on Java/Operating-System/banker.java:
```diff
@@ -0,0 +1,311 @@
+            avail[i] = 0;
+            for (int j = 0; j < p; j++) {
+                sum = sum + allocation[j][i];
+                avail[i] = res[i] - allocation[j][i];
```
@shraiyya You can add a comment here about `Checking for available resources inline`.

iamrajiv reviewed (changes requested):
@shraiyya Add space and time complexity. Also, try to give as many comments about the code.

shraiyya committed:
Update banker.java

Add a description of Banker's algorithm in the comments.
Added space and time complexity in comments.
Added more comments about the code.
Added the comment about "Checking for available resources inline."

shraiyya committed:
Update README.md

Updated link to - [Banker's Algorithm] (Operating-System/banker.java).

atarax665 committed:
Update banker.java

shraiyya commented:
Can you please label this pull request as DWOC?
Also please add the level.

shraiyya commented:
Please add the level of the PR for DWOC
